backends/sentinel/gore.go

```diff
@@ -0,0 +1,319 @@
+package sentinel
```

Comment:
Every public method must have a test. I can review the implementation only after the tests are added